core/generated-sources/src/test/kotlin/org/jetbrains/kotlinx/dataframe/samples/api/Modify.kt

```diff
@@ -1303,4 +1309,38 @@ class Modify : TestBase() {
         }
         // SampleEnd
     }
+
+    @Test
+    fun formatExample_properties() {
+        // SampleStart
+        df
+            .format().with { bold and textColor(black) }
+            .format { isHappy }.with { background(if (it) green else red) }
+            .format { weight }.notNull().linearBg(50 to FormattingDsl.blue, 90 to FormattingDsl.red)
+            .format { age }.perRowCol { row, col ->
+                textColor(
+                    linear(value = col[row], from = col.min() to blue, to = col.max() to green)
+                )
+            }
+            .toStandaloneHtml()
+        // SampleEnd
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    @Test
+    fun formatExample_strings() {
+        // SampleStart
+        df
+            .format().with { bold and textColor(black) }
+            .format("isHappy").with { background(if (it as Boolean) green else red) }
+            .format("weight").notNull().with { linearBg(it as Int, 50 to blue, 90 to red) }
+            .format("age").perRowCol { row, col ->
```

Comment:
While we do have this function, using it is worrisome. We calculate min and max (which take full scan of the column) for each row, so it's n^2 complexity out of nowhere. 

```
val col = df["age"] as DataColumn<Int>
val min = col.min()
val max = col.max()


.format("age").with { 
                textColor(
                    linear(value = col[row], from = min to blue, to = max to green)
                )
}
```

Reply:
in practice it will probably not be that bad. I wonder how many people will try to format a table with hundreds of rows, but it's good to hint towards effecient behavior